File: apps/aiyou_stack/aiyou-fastapi-services/src/autoresearch/vertex-ai-agents/examples/basic_usage.py
# Copyright (c) 2026 ShadowTag, Inc. All rights reserved.

"""Basic usage example for Vertex AI Agents
Demonstrates how to load and use agents with Vertex AI
"""

import os
import sys

from google.cloud import aiplatform
from vertexai.generative_models import GenerativeModel

# Add parent directory to path to import agent_registry
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from agent_registry import get_agent, get_agents_by_category, search_agents


def setup_vertex_ai():
    """Initialize Vertex AI"""
    aiplatform.init(
        project=os.getenv("VERTEX_AI_PROJECT_ID", "your-project-id"),
        location=os.getenv("VERTEX_AI_LOCATION", "us-central1"),
    )


def example_use_single_agent():
    """Example: Using a single agent"""
    print("=" * 60)
    print("Example 1: Using a single agent")
    print("=" * 60)

    # Get the System Architect agent
    agent = get_agent("system-architect")

    print(f"\nAgent: {agent.name}")
    print(f"Description: {agent.description}")
    print("\nCapabilities:")
    for cap in agent.capabilities:
        print(f"  - {cap}")

    # Create a Gemini model with the agent's system prompt
    model = GenerativeModel(agent.model, system_instruction=agent.system_prompt)

    # Generate a response
    user_prompt = "Analyze our codebase architecture and suggest improvements"

    response = model.generate_content(
        user_prompt,
        generation_config={
            "temperature": agent.temperature,
            "max_output_tokens": agent.max_tokens,
        },
    )

    print(f"\nUser: {user_prompt}")
    print(f"\nAgent Response:\n{response.text}")


def example_search_agents():
    """Example: Searching for agents"""
    print("\n" + "=" * 60)
    print("Example 2: Searching for agents")
    print("=" * 60)

    # Search for agents related to "API"
    results = search_agents("API")

    print(f"\nFound {len(results)} agents related to 'API':")
    for agent in results:
        print(f"\n  {agent.icon} {agent.name}")
        print(f"     {agent.description}")
        print(f"     Category: {agent.category}")


def example_browse_by_category():
    """Example: Browse agents by category"""
    print("\n" + "=" * 60)
    print("Example 3: Browse agents by category")
    print("=" * 60)

    # Get all development agents
    dev_agents = get_agents_by_category("development")

    print(f"\nDevelopment Category ({len(dev_agents)} agents):")
    for agent in dev_agents:
        print(f"\n  {agent.icon} {agent.name}")
        print(f"     {agent.description}")
        print(f"     Example: {agent.example_prompts[0]}")


def example_multi_agent_workflow():
    """Example: Multi-agent workflow"""
    print("\n" + "=" * 60)
    print("Example 4: Multi-agent workflow")
    print("=" * 60)

    # Step 1: Use System Architect to design
    architect = get_agent("system-architect")
    print(f"\n1. {architect.name}: Designing system architecture...")

    # Step 2: Use Code Refactorer to clean code
    refactorer = get_agent("code-refactorer")
    print(f"2. {refactorer.name}: Refactoring code...")

    # Step 3: Use Test Generator to add tests
    tester = get_agent("test-generator")
    print(f"3. {tester.name}: Generating tests...")

    # Step 4: Use Security Scanner to audit
    security = get_agent("security-scanner")
    print(f"4. {security.name}: Scanning for vulnerabilities...")

    print("\nMulti-agent workflow completed!")


def example_agent_with_context():
    """Example: Using an agent with conversation context"""
    print("\n" + "=" * 60)
    print("Example 5: Using an agent with conversation context")
    print("=" * 60)

    # Get the Code Mentor agent
    mentor = get_agent("code-mentor")

    # Create model with system instruction
    model = GenerativeModel(mentor.model, system_instruction=mentor.system_prompt)

    # Start a chat session
    chat = model.start_chat()

    # Multi-turn conversation
    conversations = [
        "What are SOLID principles?",
        "Can you give me an example of the Single Responsibility Principle?",
        "How would I refactor this code to follow SRP?",
    ]

    print(f"\n{mentor.icon} {mentor.name} - Teaching Session\n")

    for user_message in conversations:
        response = chat.send_message(
            user_message,
            generation_config={
                "temperature": mentor.temperature,
                "max_output_tokens": mentor.max_tokens,
            },
        )

        print(f"Student: {user_message}")
        print(f"Mentor: {response.text}\n")


def main():
    """Run all examples"""
    # Setup
    setup_vertex_ai()

    # Run examples
    try:
        example_use_single_agent()
        example_search_agents()
        example_browse_by_category()
        example_multi_agent_workflow()
        example_agent_with_context()
    except Exception as e:
        print(f"\nError running examples: {e}")
        print("Make sure you have:")
        print("  1. Set up VERTEX_AI_PROJECT_ID environment variable")
        print("  2. Authenticated with Google Cloud")
        print("  3. Installed required dependencies")


if __name__ == "__main__":
    main()

File: apps/aiyou_stack/aiyou-fastapi-services/src/autoresearch/vertex-ai-agents/examples/basic_usage.js
/**
 * Basic usage example for Vertex AI Agents
 * Demonstrates how to load and use agents with Vertex AI
 */

const { VertexAI } = require('@google-cloud/vertexai');
const { loadAgents, getAgent, searchAgents, getAgentsByCategory } = require('../index.js');

// Initialize Vertex AI
const projectId = process.env.VERTEX_AI_PROJECT_ID || 'your-project-id';
const location = process.env.VERTEX_AI_LOCATION || 'us-central1';

const vertexAI = new VertexAI({
  project: projectId,
  location: location,
});

/**
 * Example 1: Using a single agent
 */
async function exampleUseSingleAgent() {
  console.log('='.repeat(60));
  console.log('Example 1: Using a single agent');
  console.log('='.repeat(60));

  // Get the System Architect agent
  const agent = getAgent('system-architect');

  console.log(`\nAgent: ${agent.name}`);
  console.log(`Description: ${agent.description}`);
  console.log(`\nCapabilities:`);
  agent.capabilities.forEach((cap) => console.log(`  - ${cap}`));

  // Create a generative model with the agent's configuration
  const model = vertexAI.getGenerativeModel({
    model: agent.model,
    systemInstruction: agent.systemPrompt,
    generationConfig: {
      temperature: agent.temperature,
      maxOutputTokens: agent.maxTokens,
    },
  });

  // Generate a response
  const userPrompt = 'Analyze our codebase architecture and suggest improvements';

  const result = await model.generateContent(userPrompt);
  const response = result.response;

  console.log(`\nUser: ${userPrompt}`);
  console.log(`\nAgent Response:\n${response.text()}`);
}

/**
 * Example 2: Searching for agents
 */
async function exampleSearchAgents() {
  console.log('\n' + '='.repeat(60));
  console.log('Example 2: Searching for agents');
  console.log('='.repeat(60));

  // Search for agents related to "API"
  const results = searchAgents('API');

  console.log(`\nFound ${results.length} agents related to 'API':`);
  results.forEach((agent) => {
    console.log(`\n  ${agent.icon} ${agent.name}`);
    console.log(`     ${agent.description}`);
    console.log(`     Category: ${agent.category}`);
  });
}

/**
 * Example 3: Browse agents by category
 */
async function exampleBrowseByCategory() {
  console.log('\n' + '='.repeat(60));
  console.log('Example 3: Browse agents by category');
  console.log('='.repeat(60));

  // Get all development agents
  const devAgents = getAgentsByCategory('development');

  console.log(`\nDevelopment Category (${devAgents.length} agents):`);
  devAgents.forEach((agent) => {
    console.log(`\n  ${agent.icon} ${agent.name}`);
    console.log(`     ${agent.description}`);
    console.log(`     Example: ${agent.examplePrompts[0]}`);
  });
}

/**
 * Example 4: Multi-agent workflow
 */
async function exampleMultiAgentWorkflow() {
  console.log('\n' + '='.repeat(60));
  console.log('Example 4: Multi-agent workflow');
  console.log('='.repeat(60));

  // Step 1: Use System Architect to design
  const architect = getAgent('system-architect');
  console.log(`\n1. ${architect.name}: Designing system architecture...`);

  // Step 2: Use Code Refactorer to clean code
  const refactorer = getAgent('code-refactorer');
  console.log(`2. ${refactorer.name}: Refactoring code...`);

  // Step 3: Use Test Generator to add tests
  const tester = getAgent('test-generator');
  console.log(`3. ${tester.name}: Generating tests...`);

  // Step 4: Use Security Scanner to audit
  const security = getAgent('security-scanner');
  console.log(`4. ${security.name}: Scanning for vulnerabilities...`);

  console.log('\nMulti-agent workflow completed!');
}

/**
 * Example 5: Using an agent with conversation context
 */
async function exampleAgentWithContext() {
  console.log('\n' + '='.repeat(60));
  console.log('Example 5: Using an agent with conversation context');
  console.log('='.repeat(60));

  // Get the Code Mentor agent
  const mentor = getAgent('code-mentor');

  // Create model with system instruction
  const model = vertexAI.getGenerativeModel({
    model: mentor.model,
    systemInstruction: mentor.systemPrompt,
    generationConfig: {
      temperature: mentor.temperature,
      maxOutputTokens: mentor.maxTokens,
    },
  });

  // Start a chat session
  const chat = model.startChat();

  // Multi-turn conversation
  const conversations = [
    'What are SOLID principles?',
    'Can you give me an example of the Single Responsibility Principle?',
    'How would I refactor this code to follow SRP?',
  ];

  console.log(`\n${mentor.icon} ${mentor.name} - Teaching Session\n`);

  for (const userMessage of conversations) {
    const result = await chat.sendMessage(userMessage);
    const response = result.response;

    console.log(`Student: ${userMessage}`);
    console.log(`Mentor: ${response.text()}\n`);
  }
}

/**
 * Main function - Run all examples
 */
async function main() {
  try {
    // Load agents first
    await loadAgents();

    // Run examples
    await exampleUseSingleAgent();
    await exampleSearchAgents();
    await exampleBrowseByCategory();
    await exampleMultiAgentWorkflow();
    await exampleAgentWithContext();
  } catch (error) {
    console.error('\nError running examples:', error.message);
    console.log('Make sure you have:');
    console.log('  1. Set VERTEX_AI_PROJECT_ID environment variable');
    console.log('  2. Authenticated with Google Cloud');
    console.log('  3. Installed required dependencies');
  }
}

// Run if called directly
if (require.main === module) {
  main();
}

module.exports = {
  exampleUseSingleAgent,
  exampleSearchAgents,
  exampleBrowseByCategory,
  exampleMultiAgentWorkflow,
  exampleAgentWithContext,
};
